_docker/drupal/drupal-filesystem/scripts/drupal_install_checker.rb

```diff
@@ -21,148 +21,107 @@ def initialize(drupal_site, process_executor = ProcessExecutor.new, opts)
     @opts = opts
   end
 
-  def settings_exists?
-    File.exists? File.join drupal_site, 'settings.php'
-  end
+  def prepare_drupal_for_boot
+
+    wait_for_database_to_boot
+    check_all_required_drupal_configuration!
 
-  def rhd_settings_exists?
-    File.exists? File.join drupal_site, 'rhd.settings.php'
+    drush_clear_cache
+    drush_import_config
+    drush_update_db
   end
 
-  def mysql_connect?
-    begin
-      process_executor.exec! 'mysql', ["--host=#{@opts['database']['host']}",
-                                       "--port=#{@opts['database']['port']}",
-                                       "--user=#{@opts['database']['username']}",
-                                       "--password=#{@opts['database']['password']}",
-                                       '--connect-timeout=20',
-                                       "#{@opts['database']['name']}"]
-      true
-    rescue => e
-      puts "ERROR: #{e.message}"
-      false
-    end
+  private
+
+  def ensure_file_exists!(file_name)
```

Comment:
Why are you using the "!" at the end? we're not really changing the internals of the object, and it doesn't return itself.

Reply:
@LightGuard My understanding is that the exclamation mark at the end of the method name was used to denote methods in Ruby that were considered "dangerous" e.g. modified the receiver or raised an Exception. 

I've gone down the route of using the `!` if a method raises an Exception. Happy to remove this if it's not really idiomatic of Ruby as I'm still approaching a lot of this with my Java hat on :-)

Reply:
Here's what I found:

Bang (!) Methods

Ruby methods that modify an object in-place and end in an exclamation mark are known as bang methods. By convention, the bang labels a method as dangerous - specifically, as the dangerous equivalent of a method with the same name but without the bang.

You'll find a number of pairs of methods, one with the bang and one without. Those without the bang perform an action and return a freshly minted object, reflecting the results of the action (capitalizing a string, sorting an array, and so on). The bang versions of the same methods perform the action, but they do so in place: Instead of creating a new object, they transform the original object.

Examples of such pairs of methods include sort/sort! for arrays, upcase/upcase! for strings, chomp/chomp! for strings, and reverse/reverse! for strings and arrays. In each case, if you call the non-bang version of the method on the object, you get a new object. If you call the bang version, you operate in-place on the same object to which you sent the message.

In Ruby you can define a method name that ends with an exclamation point or bang. The bang methods are called and executed just like any other method. However, by convention, a method with an exclamation point or bang is considered dangerous.

Normally for the built-in classes, dangerous usually (although not always) means this method, unlike its non-bang equivalent, permanently modifies its receiver.

You'll find a number of methods, one with the bang and one without. Those without the bang perform an action and return a new object. The bang versions of the same methods perform the action, but they do so in place: Instead of creating a new object, they transform the original object.

A few non-bang methods perform changes on the original string. The names of these methods make it clear that this is happening (such as replace), even though there's no ! on the name.

Reply:
If you feel it falls in that definition, okay, cool, I can go with that.

Reply:
@LightGuard Updated to remove the `!` from methods as I don't see them being "dangerous" as per that description above.

cheers,

Rob